src/Bicep.Cli/Commands/BuildParamsCommand.cs

```diff
@@ -58,83 +58,56 @@ public async Task<int> RunAsync(BuildParamsArguments args)
                 logger.LogWarning(CliResources.ResourceTypesDisclaimerMessage);
             }
 
-            if(bicepInputPath != null && !IsBicepFile(bicepInputPath))
+            if(bicepFileArgPath != null && !IsBicepFile(bicepFileArgPath))
             {
-                throw new CommandLineException($"{bicepInputPath} is not a bicep file");
+                throw new CommandLineException($"{bicepFileArgPath} is not a bicep file");
             }
 
-            if(features.ParamsFilesEnabled && IsBicepparamsFile(paramsInputPath)) 
+            if(!features.ParamsFilesEnabled) 
             {
-                var paramsCompilation = await compilationService.CompileAsync(paramsInputPath, args.NoRestore);
+                logger.LogError(CliResources.UnableToCompileParamsFile, paramsInputPath, nameof(ExperimentalFeaturesEnabled.ParamsFiles));
+                return 1;
+            }
 
-                var bicepSemanticModel = GetBicepSemanticModelFromParamsCompilation(paramsCompilation);
+            if (!IsBicepparamsFile(paramsInputPath))
+            {
+                logger.LogError(CliResources.UnrecognizedFileExtensionMessage, paramsInputPath);
+                return 1;
+            }
 
-                if(bicepSemanticModel != null && bicepInputPath != null) 
-                {   
-                    var bicepFilePath = bicepSemanticModel.Root.FileUri;
-                    
-                    if(!bicepFilePath.Equals(PathHelper.FilePathToFileUrl(bicepInputPath)))
-                    {
-                        throw new CommandLineException($"Bicep file {bicepInputPath} provided with --bicep-file option doesn't match the Bicep file {bicepSemanticModel.Root.Name} referenced by the using declaration in the parameters file");
-                    }
+            var paramsCompilation = await compilationService.CompileAsync(paramsInputPath, args.NoRestore);
+
+            var paramsSemanticModel = paramsCompilation.GetEntrypointSemanticModel();
+
+            //Failure scenario is ignored since a diagnostic for it would be emitted during semantic analysis 
+            if(paramsSemanticModel.Root.TryGetBicepFileSemanticModelViaUsing(out var bicepSemanticModel, out _))
+            {
+                var bicepFileUsingPathUri = bicepSemanticModel.Root.FileUri;
+
+                if(bicepFileArgPath is {} && !bicepFileUsingPathUri.Equals(PathHelper.FilePathToFileUrl(bicepFileArgPath))) 
```

Comment:
Is the plan to keep this check around long term, or is it here to simplify things while params files are still under development? This check makes it effectively impossible to use the same params file with multiple templates directly in the az CLI, though users could always compile their params and bicep files in a separate step, then deploy the compiled JSON artifacts.

I think this approach is fine for now, but long term, this feels like a feature gap between Bicep and JSON parameters (especially since we know some users have multiple templates that use the same parameters file).

Reply:
The scenario where one params file is being used with multiple templates was assumed to be not as common. Hence, we introduced `using` declaration in the params specifically (multiple params files utilizing one template not vice versa). 

The `--bicep-file` arg is the source of truth in the compilation process. The source of truth is the `using` declaration which means we have to make `using` declaration optional first before we allow users to compile one params file with multiple bicep templates.

For first release at least, we plan to keep `using` declaration as required (essentially keeping this check as well)

Reply:
I agree that it's quirky. The main goal here is to get a scoped-down first release out so that we can start collecting feedback on other scenarios we need to support. The risk with starting permissive is that we'll be stuck supporting these scenarios and have less flexibility with the future design.